### ISSUE TITLE
Parallelization

### ISSUE THREAD

**Issue by FKuerten** [iteratedecks/iteratedecks#13]
I thought it would be nice to add paralleization support back into iteratedecks core.

There are different options to do. We certainly don't want any unportable solution. Thus windows threads (as used in evaluatedecks) and pthreads are probably a bad idea.

I would suggest OpenMP as it is
a) backward compatible: If you compile it with a compiler that does not support OpenMP the result will be a valid single threaded application.
b) easy to use
c) supported by many compilers, at least gcc and MS

An alternative would be boost::thread. This has more features but is also more complicated.

Any thoughts?

**Comment by MrHen** [comment]
Not my area of expertise but your suggestion sounds good to me. I have no reason to disagree.

**Comment by MrHen** [comment]
@Pr4370r14n Hey, now that people are starting to take an interest in submitting simulations to the Fansite we are probably going to get more questions about this. Are you still around/active/interested in doing this work?

http://tyrant.40in.net/kg/sim.php?sim_id=c2bd1IFw#comments:pid=16453:spid=25909

**Comment by FKuerten** [comment]
I am still around. Intend to continue working on ID thursday.

**Comment by MrHen** [comment]
@Pr4370r14n Cool. No rush/worries, I was just curious. I tried to make the branching easier to deal with but in the process feel like I made everything _way_ more complicated... so when you get back feel free to do whatever.

My current task is dealing with the collisions being reported on the Fansite:
http://tyrant.40in.net/kg/sim.php?c=collision&sim_id=c2bd1IFw

I hope to push out a 1.0.5 build later tonight but we'll see. In the process of fixing these many bugs I realized that the core engine of the game is in desperate need of some scrubbing. There is a lot of crap leftover from EvaluateDecks.
